### PR TITLE
Asset and slots

### PR DESCRIPTION
**Pull request by HKotsev**

### CODE REVIEW

File: storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/models/product/decorators/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+var base = module.superModule;
+
+var originalExports = Object.keys(module.superModule);
+originalExports.forEach(function (originalExport) {
```
Comment (isergeev19) [inline storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/models/product/decorators/index.js:4]: Why do we need this? You can just get the super module, add your shipping and export it

File: storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/models/product/decorators/shipping.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    const contentAsset = ContentMgr.getContent(
+        `product-shipping-${product.ID}`
+    );
+
+    if (contentAsset && contentAsset.online) {
+        shipping = contentAsset.custom.body;
+    }
+
+    Object.defineProperty(object, "shipping", {
```
Comment (isergeev19) [inline storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/models/product/decorators/shipping.js:16]: Is this the best name for this attribute?

File: storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/scripts/checkout/checkoutHelpers.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    module.exports[originalExport] = module.superModule[originalExport];
+});
+
+base.sendConfirmationEmail = function (order, locale) {
+    var OrderModel = require("*/cartridge/models/order");
+    var emailHelpers = require("*/cartridge/scripts/helpers/emailHelpers");
+    var Locale = require("dw/util/Locale");
+    var extractedProducts = require("*/cartridge/scripts/helpers/extractingProductsHelpers");
+
+    // var ProductFactory = require("*/cartridge/scripts/factories/product");
```
Comment (isergeev19) [inline storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/scripts/checkout/checkoutHelpers.js:17]: Commented code

File: storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/templates/default/rendering/category/catLanding.isml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<isdecorate template="common/layout/page">
```
Comment (isergeev19) [inline storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/templates/default/rendering/category/catLanding.isml:1]: You can use category in the namings, cat could be confusing

File: storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/models/product/decorators/shipping.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+"use strict";
+
+var ContentMgr = require("dw/content/ContentMgr");
+
+module.exports = function (object, product) {
+    let shipping = "";
+
+    const contentAsset = ContentMgr.getContent(
+        `product-shipping-${product.ID}`
```
Comment (isergeev19) [inline storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/models/product/decorators/shipping.js:9]: How is the product id replaced with the relevant unique info per product?

File: storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/models/product/decorators/index.js (hunk 1 above)
Comment (HKotsev) [reply storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/models/product/decorators/index.js:4]: fixed

File: storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/models/product/decorators/shipping.js (hunk 2 above)
Comment (HKotsev) [reply storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/models/product/decorators/shipping.js:16]: fixed

File: storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/scripts/checkout/checkoutHelpers.js (hunk 3 above)
Comment (HKotsev) [reply storefront-reference-architecture-master/cartridges/app_storefront_assets_slots/cartridge/scripts/checkout/checkoutHelpers.js:17]: fixed

**Pull request merged**